### PR TITLE
Fix issue when initializing DB based resource group manager

### LINKED ISSUE #22890: Issue while initializing Database based Resource Group Manager
An exception is thrown when the selectors table is being created during initialization. 

```
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'schema VARCHAR(1024),
  principal_regex VARCHAR(1024),
  id BIGINT NOT NULL AUTO' at line 10
```

## Your Environment

* Presto version used: Latest Master (0.288-snapshot)
* Data source and connector used: Resource Groups (DB based with mysql)

## Expected Behavior

Successful initialization of the required tables if they don't exist in the mysql schema

## Current Behavior

Above exception is thrown when selectors table is being created

## Possible Solution

Column `schema` is a reserved keyword, so need to surround it by backticks

### PR DESCRIPTION
**Pull request by imjalpreet**
## Description

Fixes #22890 

## Motivation and Context

This PR fixes the bug mentioned in the above issue and also updates the resource groups documentation to make it more accurate and remove any erroneous information.

## Test Plan

Verified setting up resource groups against a new database in mysql.

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by tdcmeehan** [comment]
@imjalpreet is this just a documentation fix?

File: presto-docs/src/main/sphinx/admin/resource-groups.rst [hunk 1]
```diff
@@ -213,6 +211,23 @@ Here are the key properties that can be set for a Resource Group:
 * ``jmxExport`` (optional):  If set to ``true``, the statistics of the resource group will be exported via JMX.
   Defaults to ``false``.
 
+* ``perQueryLimits`` (optional): specifies max resources each query in a
+  resource group may consume before being killed. These limits are not inherited from parent groups.
+  May set three types of limits:
+
+    * ``executionTimeLimit`` (optional): Specify an absolute value (i.e. ``1h``)
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/admin/resource-groups.rst:218]:
```suggestion
    * ``executionTimeLimit`` (optional): Specify an absolute value (for example, ``1h``)
```
This comment applies to the three sub-items in this list - please make the same changes you make to the other two items in the list (``totalMemoryLimit`` and ``cpuTimeLimit``).

* Do not use Latin abbreviations such as i.e.. For context and suggestions, see the GitLab documentation style guide recommended word list entry for [i.e.](https://docs.gitlab.com/ee/development/documentation/styleguide/word_list.html#ie) and [e.g.](https://docs.gitlab.com/ee/development/documentation/styleguide/word_list.html#eg). 

* Please change the indentation of the three items in the sublist to remove the vertical bar. Use the same indentation as the sublist in ``schedulingPolicy`` that begins with ``fair`` and ends with ``query_priority`` (``query_priority`` is visible in the top of the screenshot). 
Screenshot 2024-06-03 at 9 54 27 AM

File: presto-docs/src/main/sphinx/admin/resource-groups.rst [hunk 2]
```diff
@@ -266,6 +281,16 @@ Here are the key components of selector rules in PrestoDB:
 * ``clientTags`` (optional): List of tags. To match, every tag in this list must be in the list of
   client-provided tags associated with the query.
 
+* ``selectorResourceEstimate`` (optional): Resource Group Selection based on resource estimates.
+    - ``executionTime``
+    - ``cpuTime``
+    - ``peakMemory``
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/admin/resource-groups.rst:287]: Does ``peakMemory`` relate to ``totalMemoryLimit``? ``executionTimeLimit`` and ``executionTime``, and ``cpuTimeLimit`` and ``cpuTime`` make sense to me, but ``peakMemory`` and ``totalMemoryLimit`` not being the same I found a little confusing. 

Nit: earlier, the items are listed in the order ``executionTimeLimit``, ``totalMemoryLimit``, ``cpuTimeLimit``. 
Suggest ordering the concepts of the two lists in the same order to help the reader, for example ordering this list as ``executionTime``, ``peakMemory``, ``cpuTime``.

File: presto-docs/src/main/sphinx/admin/resource-groups.rst [hunk 3]
```diff
@@ -266,6 +281,16 @@ Here are the key components of selector rules in PrestoDB:
...
+
+* ``clientInfo`` (optional): String to match against client info.
+
+* ``principal`` (optional): This is a regular expression that matches the principal who is submitting the query.
+
+* ``schema`` (optional): This matches the session schema of the query
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/admin/resource-groups.rst:293]:
```suggestion
* ``schema`` (optional): This matches the session schema of the query.
```
Nit, punctuation.

File: presto-docs/src/main/sphinx/admin/resource-groups.rst [hunk 4]
```diff
@@ -368,26 +393,14 @@ File Resource Group Manager
...
 
-This example is for a MySQL database.
-
-.. literalinclude:: resource-groups-example1.sql
-    :language: sql
-
 .. code-block:: sql
 
     -- global properties
+    INSERT INTO resource_groups_global_properties (name, value) VALUES ('cpu_quota_period', '1h');
```
Comment (steveburnett) [inline presto-docs/src/main/sphinx/admin/resource-groups.rst:399]: No change suggested, I just wanted to say this is a good code example.

**Review by steveburnett — CHANGES_REQUESTED** [review]
Thanks for the documentation! I made a few suggestions, found a couple of nits, but this is good.

**Comment by imjalpreet** [comment]
@tdcmeehan No, one of the commits also has a small fix for the linked issue.

File: presto-docs/src/main/sphinx/admin/resource-groups.rst [hunk 5]
```diff
@@ -119,14 +119,21 @@ To set up a database-based resource group manager:
...
-    resource-groups.config-db-password=password
+    resource-groups.config-db-url=jdbc:mysql://localhost:3306/resource_groups?user=<user>&password=<password>
+
+Replace ``<user>`` and ``<password>`` with the actual username and password
 
 With the Database Resource Group Manager, changes to the configuration in the database take effect immediately
 and do not require a restart of the Presto server. This allows for more flexibility and
 dynamic changes to the resource group configurations.
 
+The resource group configuration must be populated through tables
```
[the remark this answers is not in the document]
Comment (imjalpreet) [reply presto-docs/src/main/sphinx/admin/resource-groups.rst:130]: "from the" tables would change the meaning. We want to convey that the tables created should be used to populate the resource group configuration. If you feel we can rephrase it better, please let me know.

File: presto-docs/src/main/sphinx/admin/resource-groups.rst (hunk 2 above)
Comment (imjalpreet) [reply presto-docs/src/main/sphinx/admin/resource-groups.rst:287]: > Does peakMemory relate to totalMemoryLimit? executionTimeLimit and executionTime, and cpuTimeLimit and cpuTime make sense to me, but peakMemory and totalMemoryLimit not being the same I found a little confusing.

So, they are not directly related to each other. The first section about Resource Group Properties (`executionTimeLimit`, `totalMemoryLimit`, `cpuTimeLimit`) is to set limits for a resource group. This means any query that exceeds these limits will be killed. 

The second section under selector rules (`executionTime`, `peakMemory`, `cpuTime`) is used for determining which resource group a query should be directed to. These are resource estimates that can be passed by the user as a request header(`X-Presto-Resource-Estimate`) or a command line argument(`--resource-estimates`) if the user has an idea of how much resources a query might use.

**Comment by imjalpreet** [comment]
@steveburnett Thank you for the review. I have updated the docs.

File: presto-docs/src/main/sphinx/admin/resource-groups.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (steveburnett) [reply presto-docs/src/main/sphinx/admin/resource-groups.rst:130]: No, your explanation makes sense. No change needed here, thanks!

**Review by steveburnett — APPROVED** [review]
LGTM! (docs)

Pull updated branch, reviewed the new local docs build. Everything looks good, thanks!

**Review by tdcmeehan — APPROVED** [review]
Thanks @imjalpreet

**Pull request merged**